src/axom/quest/ScatteredInterpolation.hpp

```diff
@@ -244,6 +247,107 @@ class ScatteredInterpolation
   using PointType = typename DelaunayTriangulation::PointType;
   using BoundingBoxType = typename DelaunayTriangulation::BoundingBox;
 
+private:
+  using MortonIndexType = axom::uint64;
+
+  using VertexSet = typename DelaunayTriangulation::IAMeshType::VertexSet;
+  using VertexIndirectionSet =
+    slam::CoreArrayIndirectionSet<typename VertexSet::PositionType, axom::IndexType>;
+
+private:
+  /**
+   *  \brief Helper struct for sorting input points in the Biased Randomized Incremental Order (BRIO)
+   *
+   *  BRIO helps improve worst-case performance on poorly ordered point sets.
+   *  It was introduced in the following paper:
+   *    N. Amenta, S. Choi, and G. Rote. "Incremental constructions con BRIO."
+   *    Proceedings of the 19th annual symposium on Computational geometry, 2003.
+   */
+  struct BrioComparator
+  {
+    axom::IndexType m_index;
+    int m_level;
+    MortonIndexType m_morton;
+
+    BrioComparator(IndexType index, int level, MortonIndexType morton)
+      : m_index(index)
+      , m_level(level)
+      , m_morton(morton)
+    { }
+
+    friend bool operator<(const BrioComparator& lhs, const BrioComparator& rhs)
+    {
+      return (lhs.m_level == rhs.m_level) ? lhs.m_morton < rhs.m_morton
+                                          : lhs.m_level < rhs.m_level;
+    }
+  };
+
+  /**
+   * \brief Generates a permutation of [0, pts.size()) following BRIO 
+   * 
+   * \sa BrioComparator
+   */
+  template <typename PointArray>
+  axom::Array<axom::IndexType> computeInsertionOrder(const PointArray& pts,
```

Comment:
This function reorders the input points following BRIO

Reply:
Thanks! You mentioned in an email that this results in a 20% speedup, good work!